### ISSUE TITLE
 Monthly Chat Agenda March 2024 (2024-03-04 and 2024-03-18) 

### ISSUE THREAD

**Issue by dune73** [coreruleset/coreruleset#3530]
This is the Agenda for the two Monthly CRS Chats.

The general chat is going to happen on https://owasp.slack.com in the channel #coreruleset on Monday, 2024-03-04, at 20:30 CET. That's the 1st Monday of the month. A separate issue chat is happening at the same location, same time on Monday, 2024-03-18. That's the 3rd Monday of the month. Please note that we have a [CRS calendar](https://calendar.google.com/calendar/u/0/embed?src=c_glsll5etlmcb3it0k433fcl5rc@group.calendar.google.com) (maintained by @fzipi).

Archived previous meetings and their decision are [here](https://github.com/coreruleset/project-chat-archive/blob/master/README.md).

### Outside development

* [OWASP CRS co-lead Walter Hop has died](https://coreruleset.org/20240218/the-owasp-crs-project-mourns-the-death-of-co-leader-walter-hop/)
* [Release of CRS v4](https://coreruleset.org/20240214/let-crs-4-be-your-valentine/)
* [Blog post on Early Blocking](https://coreruleset.org/20240227/new-feature-spotlight-early-blocking/)
* :rocket: Team welcomes new developer Esad Cetiner / [@EsadCetiner](https://github.com/EsadCetiner)

#### Rules

* feat: add check for combinations of t:lowercase and (?i) to lint (Franziska Bühler) [#3584]
* fix: remove t:lowercase from rules that use '(?i)' modifier in their regex (942150 PL2, 942151 PL1, 942152 PL2) (Ervin Hegedus) [#3585]
* test: change HTTP method to uppercase for test 932260-28 (Matteo Pace) [#3580]
* feat: move HTTP header rules to phase 1 (932161 PL2, 932205 PL2, 932206 PL2, 932237 PL3) (Esad Cetiner) [#3570]
* fix: prevent FPs against names due to "cron" (932260 PL1, 932236 PL2, 932237 PL3, 932239 PL2) (@superlgn) [#3578]
* chore: add Esad Cetiner to list of developers (@EsadCetiner) [#3589]
* fix: add missing tags and ver action (various rules) (Jozef Sudolský) [#3571]
* fix: adding more missing tags and ver actions (Jozef Sudolský) [#3593]

#### CRS Sandbox

* :arrows_clockwise: With Andrea leaving the project, the Sandbox has to be transferred. @fzipi is in charge and plans to deploy anew.

#### Security

* :incoming_envelope: The OWASP ModSecurity project has invited OWASP CRS to a SecLang round table to talk about URI decoding behavior across various implementations.

#### Plugins

* :mag: v4 container images revealed that plugin tests were all being run at paranoia level 1
* :arrow_up: Updated Nextcloud plugin tests and rules to work at paranoia level 4

#### Documentation and Public Relations

* :memo: Three new blog posts published (moved to _Outside development_ section above)

#### Project Administration and Sponsor relationships

* :mega: All sponsors were notified of our CRS release. The Swiss Post CISO congratulated the project on the release.
* OWASP HQ is behind with reimbursement requests. Some requests for developer retreat still not paid out, but we see movement.

#### Tools

* :test_tube: Nginx (modsec3-nginx) regression test suite remains blocked by missing go-ftw features at this time.
* :hammer_and_wrench: crs-toolchain milestone version 2.0.0 released with improvements to regular expression generation. In general, expressions will now be slightly shorter. We had to regenerate all expressions because of that but the changes were only syntactical.
#### Testing incl. Seaweed and many future plans

* FIXME: Please fill in

#### Containers

* Published v4 images two weeks after the v4 release (we were in the process of doing it and then forgot)

#### CRS Status Page

* No news yet, but time to get going

## Project discussions and decisions

* Brief explanation what is really going on with the nextcloud plugin development (huge eruption of traffic)
* Troubles with updates to Changelog PRs (takes additional to approve, which leads to delays and merge conflicts that are annoying to resolve)
* Release 4.1.0
  * LTS release?
* Do we want to allocate budget for maintenance of virtual envs (sandbox, upcoming status page)?
* Plugin changelogs: do we really need them?
* We need to discuss the order of transformations t:urlDecodeUni / t:utf8toUnicode (#3450)

### PRs that have been merged since the last meeting

* #3614
* #3613
* #3612
* #3601
* #3606
* #3604
* #3600
* #3596
* #3594
* #3599
* #3602
* #3598
* #3485
* #3593
* #3571
* #3590
* #3587
* #3589
* #3588
* #3578
* #3586
* #3572
* #3584
* #3585
* #3580
* #3574
* #3570
* #3568
* #3567
* #3548

We merged 30 PRs since the last monthly project chat.

### Open PRs marked *DRAFT* or *work in progress* or *needs action*

* #3617
* #3615
* #2637
* #3292
* #3595
* #3450

## Separate 2nd Meeting (Monday, 2024-03-18)

* 4.1.0 Release: Problems with solving issue #3610 / PR https://github.com/coreruleset/crs-toolchain/pull/137 may delay release
* We have received a report identifying ReDoS problems in several rules. Unsure if this should delay a release.

## How to get to our slack and join the meeting?

If you are not yet on the OWASP Slack, here is your invite: https://owasp.org/slack/invite .

Everybody is welcome to join our community chat.

**Comment by franbuehler** [comment]
## Decisions March 4

Troubles with updates to Changelog PRs
🔵 Decision: @theseion will look into the [GitHub feature merge queue](https://github.blog/2023-07-12-github-merge-queue-is-generally-available/) 

Release 4.1.0
🔵 Decision: feature freeze / cut-off +/- at the 2nd meeting. Release date after the 2nd chat of the month. But if we need a 4.1.1 fixing something the next day, we can.

**Comment by dune73** [comment]
# Decisions March 18

* We acknowledge the 3QA security finding. A solution will take a lot of time and there are a lot of rules affected, probably many of them in individual ways. Reporter Mirko sees various measurements:
   * 1: Having a tool which reports the problem
   * 2: look at the worst offenders.
   * 3: see if we can change some of these regexes automatically
   * 4: rely on PCRE_MATCH_LIMIT
   * 5: change to re2 and friends

Automation could be key here as well, but it takes more thinking. We are aware that a solution will take a lot of time.

* The `\v` problem in #3610 has been solved.

Under the line, there is nothing keeping back a release, so we will proceed with 4.1.0.
